File: python_leaoai/python_leaoai/models/Factory/Acabamento.py
# models/acabamento.py

from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.orm import relationship, Mapped, mapped_column
from database import Base
from .client import Client  # Certifique-se de ter o modelo Client definido

class Acabamento(Base):
    __tablename__ = "acabamento"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    nome: Mapped[str] = mapped_column(String, nullable=True)
    codigo: Mapped[str] = mapped_column(String, default="", nullable=True)
    imagem: Mapped[str] = mapped_column(String, default="", nullable=True)
    imagem_formulacao: Mapped[str] = mapped_column(String, default="", nullable=True)
    ativo: Mapped[bool] = mapped_column(Boolean, default=True)
    excluido: Mapped[bool] = mapped_column(Boolean, default=False)

    client_id: Mapped[int] = mapped_column(Integer, ForeignKey("client.id"))
    client: Mapped[Client] = relationship("Client", back_populates="acabamentos")

File: Framework.Data/Models/Factory/Acabamento.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Framework.Data.Models.Factory
{
    public class Acabamento
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Codigo { get; set; } = "";
        public string Imagem { get; set; } = "";
        public string ImagemFormulacao { get; set; } = "";
        public bool Ativo { get; set; }
        public bool Excluido { get; set; }
        [ForeignKey("Client")]
        public int ClientId { get; set; }
        public virtual Client Client { get; set; }
    }
}
